x/incentive/keeper/keeper_withdraw.go

```diff
@@ -0,0 +1,154 @@
+
+		//-----------------------------
+		// Dex rewards commission
+		//-----------------------------
+		// to give = delegated amount / total delegation * newly minted eden * commission rate
+		dexRewardsCommission := delegatedAmt.QuoInt(totalDelegationAmt).Mul(dexRewards).Mul(comm_rate)
+		// Sum total commission given
+		totalDexRewardsGiven = totalDexRewardsGiven.Add(dexRewardsCommission.TruncateInt())
+		//-----------------------------
+
+		// increase uncomitted token amount of validator's commitment
+		k.UpdateTokensForValidator(ctx, valAddr.String(), edenCommission.TruncateInt(), dexRewardsCommission)
+
+		return false
+	})
+
+	return totalEdenGiven, totalDexRewardsGiven
+}
+
+// withdraw rewards
+// Eden, EdenBoost and Elys to USDC
+func (k Keeper) ProcessWithdrawRewards(ctx sdk.Context, delegator string) error {
+	_, err := sdk.AccAddressFromBech32(delegator)
+	if err != nil {
+		return err
+	}
+
+	// Get commitments
+	commitments, bfound := k.cmk.GetCommitments(ctx, delegator)
+	if !bfound {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "unable to find commitment")
+	}
+
+	// Eden
+	uncommittedEden, _ := commitments.GetUncommittedTokensForDenom(ptypes.Eden)
+	// Eden B
+	uncommittedEdenB, _ := commitments.GetUncommittedTokensForDenom(ptypes.EdenB)
+	// Elys
+	uncommittedElys, _ := commitments.GetUncommittedTokensForDenom(ptypes.Elys)
+
+	// Withdraw Eden
+	err = k.cmk.ProcessWithdrawTokens(ctx, delegator, ptypes.Eden, uncommittedEden.Amount)
+
+	// Withdraw Eden Boost
+	err = k.cmk.ProcessWithdrawTokens(ctx, delegator, ptypes.EdenB, uncommittedEdenB.Amount)
+
+	// Convert Elys to USDC
+	// Conversion is done inside commmitment module, later on we will have swap module
+	err = k.cmk.ProcessWithdrawElysTokens(ctx, delegator, ptypes.Elys, uncommittedElys.Amount)
+
+	return err
+}
+
+// withdraw validator commission
+// Eden, EdenBoost and Elys to USDC
+func (k Keeper) ProcessWithdrawValidatorCommission(ctx sdk.Context, delegator string, validator string) error {
+	_, err := sdk.AccAddressFromBech32(delegator)
+	if err != nil {
+		return err
+	}
+
+	// Check validator address
+	_, err = sdk.ValAddressFromBech32(validator)
+	if err != nil {
+		return err
+	}
+
+	// Get commitments
+	commitments, bfound := k.cmk.GetCommitments(ctx, validator)
+	if !bfound {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "unable to find commitment")
+	}
+
+	// Eden
+	uncommittedEden, _ := commitments.GetUncommittedTokensForDenom(ptypes.Eden)
+	// Eden B
+	uncommittedEdenB, _ := commitments.GetUncommittedTokensForDenom(ptypes.EdenB)
+	// Elys
+	uncommittedElys, _ := commitments.GetUncommittedTokensForDenom(ptypes.Elys)
+
+	// Withdraw Eden
+	err = k.cmk.ProcessWithdrawValidatorCommission(ctx, delegator, validator, ptypes.Eden, uncommittedEden.Amount)
+
+	// Withdraw Eden Boost
+	err = k.cmk.ProcessWithdrawValidatorCommission(ctx, delegator, validator, ptypes.EdenB, uncommittedEdenB.Amount)
+
+	// Convert Elys to USDC
```

Comment:
Can you add TODO here? (I think swap logic is missing implementation here) - Or if it's not relevant on this section, you can remove the comment

Reply:
Okay got it. Will do this in the next PR.